.github/workflows/faucet_test.yml

```
@@ -0,0 +1,58 @@
+name: Faucet Tests
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  faucet-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+
+    strategy:
```

```suggestion
    strategy:
      max-parallel: 1
```
Its better to execute these tests one a time to avoid being rate limited.